lib/openid_connect/worker.ex

```diff
@@ -11,21 +11,31 @@ defmodule OpenIDConnect.Worker do
 
   require Logger
 
-  def start_link(provider_configs, name \\ :openid_connect) do
-    GenServer.start_link(__MODULE__, provider_configs, name: name)
+  def start_link(options, name \\ :openid_connect)
+
+  def start_link({provider_configs, notify_pid}, name) do
+    GenServer.start_link(__MODULE__, {provider_configs, notify_pid}, name: name)
+  end
+
+  def start_link(provider_configs, name) do
+    start_link({provider_configs, nil}, name)
   end
 
-  def init(:ignore) do
+  def init({:ignore, _pid}) do
     :ignore
   end
 
-  def init(provider_configs) do
+  def init({provider_configs, notify_pid}) do
```

Comment:
I think instead this should be a separate functional call that is something like:

```elixir
def init({provider_configs, pid}) is_id(pid) do
   {:ok, state} = init(provider_configs)
   Process.send(pid, :ready)
   {:ok, state}
end
```

you can drop the `handle_info` function

Reply:
Third time lucky. I'm using `Process.send_after` rather than `Process.send` to ensure that update_documents will be in the workers inbox before the test proceeds.